dev/AppNotifications/AppNotificationBuilder/AppNotificationBuilder.cpp

```diff
@@ -0,0 +1,372 @@
+{
+    
+
+    std::wstring AppNotificationBuilder::GetWinSoundEventString(AppNotificationSoundEvent soundEvent)
+    {
+        static std::map<AppNotificationSoundEvent, std::wstring> c_soundEventMap
+        {
+            { AppNotificationSoundEvent::Default, L"ms-winsoundevent:Notification.Default"},
+            { AppNotificationSoundEvent::IM, L"ms-winsoundevent:Notification.IM"},
+            { AppNotificationSoundEvent::Mail, L"ms-winsoundevent:Notification.Mail"},
+            { AppNotificationSoundEvent::Reminder, L"ms-winsoundevent:Notification.Reminder"},
+            { AppNotificationSoundEvent::SMS, L"ms-winsoundevent:Notification.SMS"},
+            { AppNotificationSoundEvent::Alarm, L"ms-winsoundevent:Notification.Looping.Alarm"},
+            { AppNotificationSoundEvent::Alarm2, L"ms-winsoundevent:Notification.Looping.Alarm2"},
+            { AppNotificationSoundEvent::Alarm3, L"ms-winsoundevent:Notification.Looping.Alarm3"},
+            { AppNotificationSoundEvent::Alarm4, L"ms-winsoundevent:Notification.Looping.Alarm4"},
+            { AppNotificationSoundEvent::Alarm5, L"ms-winsoundevent:Notification.Looping.Alarm5"},
+            { AppNotificationSoundEvent::Alarm6, L"ms-winsoundevent:Notification.Looping.Alarm6"},
+            { AppNotificationSoundEvent::Alarm7, L"ms-winsoundevent:Notification.Looping.Alarm7"},
+            { AppNotificationSoundEvent::Alarm8, L"ms-winsoundevent:Notification.Looping.Alarm8"},
+            { AppNotificationSoundEvent::Alarm9, L"ms-winsoundevent:Notification.Looping.Alarm9"},
+            { AppNotificationSoundEvent::Alarm10, L"ms-winsoundevent:Notification.Looping.Alarm10"},
+            { AppNotificationSoundEvent::Call, L"ms-winsoundevent:Notification.Looping.Call"},
+            { AppNotificationSoundEvent::Call2, L"ms-winsoundevent:Notification.Looping.Call2"},
+            { AppNotificationSoundEvent::Call3, L"ms-winsoundevent:Notification.Looping.Call3"},
+            { AppNotificationSoundEvent::Call4, L"ms-winsoundevent:Notification.Looping.Call4"},
+            { AppNotificationSoundEvent::Call5, L"ms-winsoundevent:Notification.Looping.Call5"},
+            { AppNotificationSoundEvent::Call6, L"ms-winsoundevent:Notification.Looping.Call6"},
+            { AppNotificationSoundEvent::Call7, L"ms-winsoundevent:Notification.Looping.Call7"},
+            { AppNotificationSoundEvent::Call8, L"ms-winsoundevent:Notification.Looping.Call8"},
+            { AppNotificationSoundEvent::Call9, L"ms-winsoundevent:Notification.Looping.Call9"},
+            { AppNotificationSoundEvent::Call10, L"ms-winsoundevent:Notification.Looping.Call10"},
+        };
+
+        return c_soundEventMap[soundEvent];
+    }
+
+    winrt::Microsoft::Windows::AppNotifications::Builder::AppNotificationBuilder AppNotificationBuilder::AddArgument(hstring const& key, hstring const& value)
+    {
+        THROW_HR_IF(E_INVALIDARG, key.empty());
+
+        m_arguments.Insert(key, value);
```

Comment:
Locks? Specifically a write lock

Reply:
Why? `AppNotificationBuilder` is an instance of an inproc object (like std::wstring). Why would you expect 2 threads to concurrently access the same builder instance? One doesn't expect that for std::string, std::vector and zillions other types.